src/state/fairlight.ts

```diff
@@ -7,37 +7,100 @@ import {
 	FairlightInputType,
 } from '../enums'
 
-// export interface FairlightAudioMasterChannel {
-// 	equalizerEnabled: boolean
-// 	equalizerGain: number
-// 	makeUpGain: number
-// 	/** Gain in decibel, -Infinity to +6dB */
-// 	faderGain: number
-// 	followFadeToBlack: boolean
-// }
+export interface FairlightAudioDynamicsState {
+	// makeUpGain: number
+
+	limiter?: FairlightAudioLimiterState
+	compressor?: FairlightAudioCompressorState
+	expander?: FairlightAudioExpanderState
+}
+
+export interface FairlightAudioLimiterState {
+	limiterEnabled: boolean
+	threshold: number
+	attack: number
+	hold: number
+	release: number
+}
+
+export interface FairlightAudioCompressorState {
+	compressorEnabled: boolean
+	threshold: number
+	ratio: number
+	attack: number
+	hold: number
+	release: number
+}
+
+export interface FairlightAudioExpanderState {
+	expanderEnabled: boolean
+	gateEnabled: boolean
+	threshold: number
+	range: number
+	ratio: number
+	attack: number
+	hold: number
+	release: number
+}
+
+export interface FairlightAudioEqualizerBandState {
+	bandEnabled: boolean
+
+	readonly supportedShapes: number[] // TODO
+	shape: number // TODO
+
+	readonly supportedFrequencyRanges: number[] // TODO
+	frequencyRange: number // TODO
+
+	frequency: number
+
+	gain: number
+	qFactor: number
+}
+
+export interface FairlightAudioMasterChannel {
+	equalizerEnabled: boolean
+	equalizerGain: number
+	readonly equalizerBands: Array<FairlightAudioEqualizerBandState | undefined>
+
+	makeUpGain: number
+	/** Gain in decibel, -Infinity to +6dB */
+	faderGain: number
+	followFadeToBlack: boolean
+
+	dynamics?: Omit<FairlightAudioDynamicsState, 'expander'>
+}
+
+export interface FairlightAudioMonitorChannel {
+	gain: number
+	inputMasterGain: number
+	inputTalkbackGain: number
+	inputSidetoneGain: number
+}
 
 export interface FairlightAudioSource {
 	properties?: FairlightAudioSourceProperties
+
+	dynamics?: FairlightAudioDynamicsState
 }
 export interface FairlightAudioSourceProperties {
 	readonly sourceType: FairlightAudioSourceType
 
 	readonly maxFramesDelay: number
 	framesDelay: number
 
-	// TODO - mono vs stereo
-
-	gain: number
-
 	readonly hasStereoSimulation: boolean
 	stereoSimulation: number
 
-	readonly equalizerBands: number
+	gain: number
+	balance: number
+	faderGain: number
+
 	equalizerEnabled: boolean
 	equalizerGain: number
+	readonly equalizerBands: Array<FairlightAudioEqualizerBandState | undefined>
+
 	makeUpGain: number
```

Comment:
Makes more sense to have it in `FairlightAudioDynamicsState` to me

Reply:
I agree, but it makes the state updates from some commands a bit more complex, as the makeUpGain is part of the general source/master command.

Ill have a play to see how messy it becomes, it might not be as bad as I remember

Reply:
yeah its a bit fiddly, but not too bad really.
In atem-connection it is ok, the state updates simply become more complex to spread the properties around.

It mostly comes down to handling it in atem-state, where it has to do the reverse before doing the diffing. But again, not too bad so im going to go with it so we have a cleaner state